### PR TITLE
Fixed previous check on speed continuity, added check max accel and f…

### PR DESCRIPTION
**Pull request by lromor**
…eedrate.

### CODE REVIEW

File: src/planner_test.cc [hunk 1]
```diff
@@ -194,9 +204,42 @@ class PlannerHarness {
...
+      const double accel =
+        fabs_accel(segment.v0, segment.v1, segment.steps[axis]);
+      const double max_accel =
+        config.acceleration[axis] * config.steps_per_mm[axis];
+      const double max_feedrate =
+        config.max_feedrate[axis] * config.steps_per_mm[axis];
+      EXPECT_LE(segment.v0, max_feedrate);
+      EXPECT_LE(segment.v1, max_feedrate);
+      if (segment.v0 != segment.v1)
+        EXPECT_NEAR_REL(accel, max_accel, rel_error);
```
Comment (lromor) [inline src/planner_test.cc:225]: Maybe let's test with `EXPECT_LE`?

File: src/planner_test.cc [hunk 2]
```diff
@@ -194,9 +204,42 @@ class PlannerHarness {
...
+    }
+  }
+}
+
+size_t GetDefiningMotor(const LinearSegmentSteps &segment) {
```
Comment (hzeller) [inline src/planner_test.cc:230]: maybe make this `static`

File: src/planner_test.cc [hunk 3]
```diff
@@ -209,9 +252,14 @@ static void VerifyCommonExpectations(
   // The joining speeds between segments match.
   for (size_t i = 0; i < segments.size() - 1; ++i) {
     // Let's determine the speed in euclidian space.
+    const size_t m1 = GetDefiningMotor(segments[i]);
+    const size_t m2 = GetDefiningMotor(segments[i + 1]);
+    if (m1 == m2) {
+      EXPECT_EQ(segments[i].v1, segments[i + 1].v0)
+        << "Joining speed between " << i << " and " << (i + 1);
+    }
 #if 1
```
Comment (hzeller) [inline src/planner_test.cc:261]: I guess we don't need the `#if 1` around here anymore, just leave it always on. It looks like it was `#if`-ed as there were reasons why it failed before.

**Review by hzeller — COMMENTED** [review]
LGTM, two smallish remarks.

File: src/planner_test.cc (hunk 1 above)
Comment (hzeller) [reply src/planner_test.cc]: Yes, as we only want to know that we never exceed the acceleration, but smaller is ok (in particular as we'll have jerk handling at some point, in which we expect it to be slower initially)

Comment (lromor) [reply src/planner_test.cc]: (solved)

File: src/planner_test.cc (hunk 2 above)
Comment (lromor) [reply src/planner_test.cc]: (solved)

File: src/planner_test.cc (hunk 3 above)
Comment (lromor) [reply src/planner_test.cc]: (solved)

**Pull request merged**